api/ipam/ipstate/ipstate.proto

```diff
@@ -0,0 +1,23 @@
+
+syntax = "proto3";
+
+import "github.com/cisco-app-networking/nsm-nse/api/ipam/ipprovider/ipprovider.proto";
+import "github.com/cisco-app-networking/nsm-nse/api/ipam/ipreceiver/ipreceiver.proto";
+
+package ippool;
+option go_package = "ipstate";
+
+
+
+service IpamStateInterrogator {
+    rpc GetAllocatedSubnets (Empty) returns (SubnetsState) {} // Request all allocated subnets
```

Comment:
I would add  methods to get the addresses by connectivity domain, and for specific NSE instances. I we deploy a wcmd instance per connectivity domain, but it is capable of supporting multiple.

Reply:
Actually, we thought about that (to get the addresses per connectivity domain), but we figured out that all the entries are stored in the same etcd. The response also contains the connectivity domain for a subnet and the name of the NSE. Do you think that it's better to get entries per connectivity domain since we can interrogate an IPAM server for other connectivity domains (different from the one that the IPAM server is deployed for)?

Reply:
The general way to make it would be to make a query API that would take some form of rangeIdentifier criteria as input (or a list of rangeIdentifiers).  The implementation of the API by a specific IPAM server should be only handling the scope of that IPAM server (single connectivity domain in our case).